### PR TITLE
(#809) Docsite Navigation

### LINKED ISSUE #809: Build Out Main Navigation on the Documentation Site
The main navigation on the documentation site needs to be built out based on the sitemap:
[NCIDS for All - Sitemap_2023-08-22.pdf](https://github.com/NCIOCPL/ncids/files/12420374/NCIDS.for.All.-.Sitemap_2023-08-22.pdf) [Updated 08/22/23]
- Archived (original site map for reference): [NCIDS for All - Sitemap.pdf](https://github.com/NCIOCPL/ncids/files/12162037/NCIDS.for.All.-.Sitemap.pdf)

### Resources:
* Current Page: https://designsystem-dev.cancer.gov/develop/
* Based off the USWDS Page: https://designsystem.digital.gov/components/overview/

### Requirements
- The side navigation shall display on pages that contain sub-pages
- When the side navigation is displayed, it shall only display sub-pages 1 level below the current page
- When a sub-page contains sub-pages at levels greater than 1, then the sub-pages shall be able to be viewed by the user by expanding the Level 1 sub-page side nav tab.
- When a sub-page contains sub-pages at levels greater than 1, the Level 1 sub-page shall have an arrow on the tab to indicate that there are deeper levels of sub-pages contained. 

### Sub-Tasks
- [x] #814 
- [x] #895

### Site content
While the site navigation and basic pages will be created in this ticket, the content on the different pages will be created in separate tickets.
- [ ] #945
- [x] #938 
- [x] #944 
- [ ] #934
- [x] #936 
- [ ] #935 
- [x] #944 
- [x] #993 
- [ ] #943 
- [ ] Create Home Page - Task #9999

### Notes
*Per meeting on 14MAR2023*
* The goal for this Enabler is to make the documentation site better for internal testing use first and then continuing to build it out so it can be the documentation site for NCIDS R1. 
* This ticket will contain all of the tasks that are/will be determined to be updated on the Documentation site.

### PR DESCRIPTION
**Pull request by karlikpj**
Closes #809

### CODE REVIEW

File: docs/src/components/__tests__/SideNavigation.test.js [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+				path: '/about/development/index.mdx',
+				label: 'Development',
+			},
+		],
+		path: '/about/index.mdx',
+		label: 'About NCIDS',
+	},
+];
+describe('Header', () => {
+	it('renders header', () => {
```
Comment (bryanpizzillo) [inline docs/src/components/__tests__/SideNavigation.test.js:63]: Please add a test for determining that the order was drawn correctly.

File: docs/src/components/__tests__/TopNavigation.test.js [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+						name: 'autocomplete',
+						path: '/components/autocomplete',
+						label: 'Autocomplete',
+						children: [],
+					},
+				],
+			},
+		];
+		const pathSplit = ['components'];
+		render(<TopNavigation data={data} path={pathSplit} />);
```
Comment (bryanpizzillo) [inline docs/src/components/__tests__/TopNavigation.test.js:49]: There should be a test here that makes sure the order is correct. (i.e. more than 1 component.)

File: docs/src/components/footer/nci-big.jsx [hunk 3]
```diff
@@ -47,22 +47,15 @@ export const NCIBigFooter = () => {
 												} `}
```
Comment (bryanpizzillo) [inline docs/src/components/footer/nci-big.jsx:47]: Why is this file being modified? This does not seem related to the navigation ticket, so please remove.

File: docs/src/components/layouts/findObjectByKey.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
+const findObjectByKey = (obj, key, value) => {
```
Comment (bryanpizzillo) [inline docs/src/components/layouts/findObjectByKey.js:1]: This should go into `docs/src/utils`

File: docs/src/components/typography/font-family-table.jsx [hunk 5]
```diff
@@ -19,7 +19,7 @@ const FontFamilyTable = ({ children }) => {
 };
```
Comment (bryanpizzillo) [inline docs/src/components/typography/font-family-table.jsx:19]: Why is this changing?

File: docs/src/components/typography/font-size-theme-table.jsx [hunk 6]
```diff
@@ -21,7 +21,7 @@ const FontSizeThemeTable = ({ children }) => {
 };
```
Comment (bryanpizzillo) [inline docs/src/components/typography/font-size-theme-table.jsx:21]: Why is this changing?

File: docs/src/components/typography/font-weight-table.jsx [hunk 7]
```diff
@@ -20,7 +20,7 @@ const FontWeightTable = ({ children }) => {
 };
 
 FontWeightTable.propTypes = {
```
Comment (bryanpizzillo) [inline docs/src/components/typography/font-weight-table.jsx:22]: Why is this changing?

File: docs/src/hooks/use-site-metadata.js [hunk 8]
```diff
@@ -2,7 +2,7 @@ import { useStaticQuery, graphql } from 'gatsby';
 
 function useSiteMetadata() {
 	const data = useStaticQuery(graphql`
-		{
+		query {
```
Comment (bryanpizzillo) [inline docs/src/hooks/use-site-metadata.js:5]: Why is this changing from before? i.e. how did it work previously without the query???

Comment (dlescarbeau) [reply docs/src/hooks/use-site-metadata.js:5]: I changed it for consistency between the two queries, and confirmed the change doesn't affect anything.

**Review by bryanpizzillo — CHANGES_REQUESTED** [review]
From a content perspective a couple of things:
1. The USWDS Components page should have the list of components on it. (These *were* on the old home page.)
2. Character Count and Memorable Date should be removed from the Components nav and placed on the USWDS Components list. They are pass throughs and pass throughs go on the USWDS Component page.

**Review by bryanpizzillo — CHANGES_REQUESTED** [review]
Sorry to ask again, but could you please change `label` to `nav_label`? (Which is what I keep asking for, but I must mumble the nav part...) However, I will just type it to make it more official for all.

**Review by bryanpizzillo — CHANGES_REQUESTED** [review]
Header.jsx dropped a LARGE amount in code coverage. It was 100% across the board, now it is:


| File                    | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s |
------------------------|---------|----------|---------|---------|-------------------
| header.jsx                |   66.67 |       50 |   33.33 |   66.67 | 26-39 |

Please get the lines covered to 100 (or close to it) and the other numbers over at least 85%.

@olitharp-nci would have told you this, but I have not finished the PR that shows code coverage.

**Review by andyvanavery31 — APPROVED** [review]
Approved by Product on 30AUG2023 in relation to the Site map last updated on 22AUG2023. Any future changes will be handled via Change Request.

File: docs/content/components/promoblock.mdx
Comment (olitharp-nci) [inline docs/content/components/promoblock.mdx:1]: rename file to promo-block.mdx?

File: docs/content/components/nci-hero.mdx
Comment (olitharp-nci) [inline docs/content/components/nci-hero.mdx:1]: rename file to hero.mdx

File: docs/content/components/nci-hero.mdx [hunk 9]
```diff
@@ -1,5 +1,7 @@
 ---
 title: NCI Hero Component - NCI Design System
```
Comment (olitharp-nci) [inline docs/content/components/nci-hero.mdx:2]: Hero Component

**Review by olitharp-nci — CHANGES_REQUESTED** [review]

File: docs/content/components/sidenav.mdx
Comment (olitharp-nci) [inline docs/content/components/sidenav.mdx:1]: should we change this to side-navigation to match uswds?

File: docs/src/scss/_sidenav.scss
Comment (olitharp-nci) [inline docs/src/scss/_sidenav.scss:1]: uh what's this for?

File: packages/ncids-css/packages/usa-sidenav/internal/_nci-sidenav.scss [hunk 10]
```diff
@@ -46,30 +46,41 @@
...
-		&::after {
-			content: '';
-			margin: 0;
-			width: 1.5rem;
-			height: 1.5rem;
-			position: absolute;
-			right: 0;
-			top: 50%;
-			transform: translateY(-50%);
+		@include at-media('desktop') {
```
Comment (olitharp-nci) [inline packages/ncids-css/packages/usa-sidenav/internal/_nci-sidenav.scss:49]: what's this change?

**Review by olitharp-nci — CHANGES_REQUESTED** [review]

File: docs/src/scss/_sidenav.scss
Comment (dlescarbeau) [reply docs/src/scss/_sidenav.scss:1]: nci-current is used as a style for the parent level of the sidenav you're at, while usa-current is for the current location of the sidenav. there is logic based on these values determining the level of the navigation

File: docs/content/components/sidenav.mdx
Comment (dlescarbeau) [reply docs/content/components/sidenav.mdx:1]: sure, why not !

File: docs/src/scss/_sidenav.scss
Comment (dlescarbeau) [reply docs/src/scss/_sidenav.scss:1]: I'll play with ripping it out and just using usa-current

File: packages/ncids-css/packages/usa-sidenav/internal/_nci-sidenav.scss (hunk 10 above)
Comment (dlescarbeau) [reply packages/ncids-css/packages/usa-sidenav/internal/_nci-sidenav.scss]: Gone

**Review by olitharp-nci — APPROVED** [review]
